src/app/custom-keyboard/_components/KeyCapOption.tsx

```diff
@@ -0,0 +1,160 @@
+'use client';
+
+import classNames from 'classnames/bind';
+import { HexColorPicker } from 'react-colorful';
+import { useCallback, useContext, useEffect, useState } from 'react';
+import { KeyColorContext, KeyboardDataContext } from '@/context/customKeyboardContext';
+import { CustomKeyboardKeyTypes } from '@/types/CustomKeyboardTypes';
+import { Color } from '@react-three/fiber';
+import styles from './KeyCapOption.module.scss';
+import ColorTag from './ColorTag';
+
+const cn = classNames.bind(styles);
+
+const HELP_CONTENT = {
+  '세트 구성': { content: 'W, A, S, D, 방향키, Esc, Enter, Space Bar', cost: '(+5,500원의 추가요금이 있습니다)' },
+  '내 맘대로 바꾸기': {
+    content: '원하는 색상으로 바꾸고 싶은 키를 선택해주세요',
+    cost: '(개당 +500원의 추가요금이 있습니다)',
+  },
+};
+
+export default function KeyCapOption() {
+  const {
+    keyboardData: { baseKeyColor, hasPointKeyCap, pointKeyType, individualColor, pointKeyColor },
+    updateData,
+    updateIndividualColor,
+    deleteIndividualColor,
+  } = useContext(KeyboardDataContext);
+  const { focusKey, updateFocusKey } = useContext(KeyColorContext);
+  const [pointColor, setPointColor] = useState(baseKeyColor);
+  const [colorList, setColorList] = useState<[CustomKeyboardKeyTypes, Color][]>(
+    Object.keys(individualColor).map((key) => [
+      key as CustomKeyboardKeyTypes,
+      individualColor[key as CustomKeyboardKeyTypes] as Color,
+    ]),
+  );
+
+  const handleChangeBaseColor = (value: Color) => {
+    updateData('baseKeyColor', value);
+  };
+
+  const handleClickPointKeyCapButton = (value: boolean) => {
+    updateData('hasPointKeyCap', value);
+    updateFocusKey(null);
+    if (value) {
+      if (pointKeyType === '세트 구성') {
+        setPointColor(pointKeyColor);
+      } else {
+        setPointColor(baseKeyColor);
+      }
+    }
```

Comment:
혹시 이렇게 하면 eslint에 안걸리는지는 모르겠지만 한 번 의견만이라두,,, 이렇게 줄일 수도 있을 것 같다,,,는 정도임니다!

```suggestion
   setPointColor(value && pointKeyType === '세트 구성' ? pointKeyColor : baseKeyColor);
```